File: frontend/src/interfaces/dispatch.ts
export interface Dispatch {
  id: string;
  originType: string;
  originId: number;
  destinationType: string;
  destinationId?: number | null;
  dispatchStatus: string;
  transportModeType: string;
  trackingNumber?: string | null;
  dispatchDate?: Date | null;
  deliveryDate?: Date | null;
  userId: number;
  orderIds: number[];
}

export interface DispatchDetail {
  id: number;
  originType: string;
  originId: number;
  destinationType: string;
  destinationId: number | null;
  dispatchStatus: string;
  transportModeType: string;
  trackingNumber: string | null;
  dispatchDate: Date | null;
  deliveryDate: Date | null;
  userId: number;
  orderNumbers: string[];
}

export interface DispatchRequest {
  destinationType: 'Warehouse' | 'Vessel' | 'Supplier' | 'Address' | null;
  destinationId: number | null;
  transportModeType: 'Air' | 'Sea' | 'Courier';
  status: 'Created' | 'Sent' | 'Delivered';
  userId: number;
  orderIds: number[];
}

File: backend/SpareHub/Domain/Models/Dispatch.cs

namespace Domain.Models;

public class Dispatch
{
    public string Id { get; set; } = null!;
    public string? OriginType { get; set; }
    public int OriginId { get; set; }
    public string? DestinationType { get; set; }
    public int? DestinationId { get; set; }
    public string DispatchStatus { get; set; } = null!;
    public string TransportModeType { get; set; } = null!;
    public string? TrackingNumber { get; set; }
    public DateTime? DispatchDate { get; set; }
    public DateTime? DeliveryDate { get; set; }
    public int UserId { get; set; }
    public ICollection<Order> Orders { get; set; } = new List<Order>();
}
